ibm/config.go

```diff
@@ -2738,6 +2752,39 @@ func (c *Config) ClientSession() (interface{}, error) {
 			"X-Original-User-Agent": {fmt.Sprintf("terraform-provider-ibm/%s", version.Version)},
 		})
 	}
+	
+	//COMPLIANCE Service v2 version
+	// Construct an "options" struct for creating the service client.
+	var postureManagementClientURLv2 string
+	if c.Visibility == "public" || c.Visibility == "public-and-private" {
+		postureManagementClientURLv2, err = posturemanagementv2.GetServiceURLForRegion(c.Region)
+	} else {
+		session.findingsClientErr = fmt.Errorf("Error occurred while configuring Security Insights Findings API service: `%v` visibility not supported", c.Visibility)
+	}
+	if err != nil {
+		postureManagementClientURLv2 = posturemanagementv2.DefaultServiceURL
```

Comment:
It should return error instead of default url, so that user ll know that region is not supported.. instead of silently creating in default region

```
postureManagementClientURLv2 := posturemanagementv2.DefaultServiceURL
	if c.Visibility == "public" || c.Visibility == "public-and-private" {
		postureManagementClientURLv2, err = posturemanagementv2.GetServiceURLForRegion(c.Region)
		if err != nil {
		postureManagementClientErrv2 = fmt.Errorf("posture management is not supported in given region %s",c.Region)
	}
	} else {
		session.postureManagementClientErrv2 = fmt.Errorf("Error occurred while configuring Security Insights Findings API service: `%v` visibility not supported", c.Visibility)
	}
	
```